apps/root/src/pages/aggregator/swap-container/index.tsx

```diff
@@ -69,32 +79,28 @@ const SwapContainer = () => {
   }, [mappedNetworks]);
 
   React.useEffect(() => {
-    if (!from) {
-      if (fromParamToken) {
-        dispatch(setFrom(fromParamToken));
-      } else if (fromParamCustomToken && !from) {
-        dispatch(setFrom(fromParamCustomToken.token));
-      } else if (!from && !to && !toParamToken && !toParamCustomToken) {
-        let networkToUse = find(mappedNetworks, { chainId: Number(chainId) });
-        if (!networkToUse && chainId) {
-          networkToUse = find(mappedNetworks, { name: chainId.toLowerCase() });
-        }
-        dispatch(
-          setFrom(
-            getProtocolToken(Number(networkToUse?.chainId || actualCurrentNetwork.chainId || currentNetwork.chainId))
-          )
-        );
-      }
+    const handleCustomTokens = (tokenAddress: Address) => {
+      void handleCustomTokenBalances({ tokenAddress, chainId: networkToUse });
```

Comment:
This network should actually be the chainId passed by param right?

Reply:
Yes, but as the param can be the network name, i've calculated it on 'networkToUse' useMemo